app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/runStats/RunInfoStatsFragment.kt

```diff
@@ -0,0 +1,278 @@
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentTransaction
+import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.machineLearning.DigitalInk
+import com.epfl.drawyourpath.map.MapFragment
+import com.epfl.drawyourpath.path.Run
+import com.epfl.drawyourpath.pathDrawing.PathDrawingDetailPerformanceFragment
+import com.epfl.drawyourpath.utils.Utils
+import com.google.android.gms.maps.model.LatLng
+import com.google.mlkit.vision.digitalink.Ink
+
+/**
+ * This fragment is used to display some information and statistics relative to a run.
+ *
+ */
+class RunInfoStatsFragment(private val run: Run) : Fragment(R.layout.fragment_run_info_stats) {
+    private lateinit var titleText: TextView
+    private lateinit var changeLeftButton: TextView
+    private lateinit var changeRightButton: Button
+
+    private lateinit var currentStateView: RunInfoStatesEnum
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        // init the elements of the view
+        initViewElements(view)
+        // at the initial state display the path drawn information
+        showPathDrawn()
+        // defined the transition of the right button
+        rightTransitionButton()
+        // defined the transition of the left button
+        leftTransitionButton()
+    }
+
+    /**
+     * Helper functions to defined the transitions made by clicking on right transition button.
+     */
+    private fun rightTransitionButton() {
+        this.changeRightButton.setOnClickListener {
+            when (currentStateView) {
+                RunInfoStatesEnum.PATH_DRAWN -> showGlobalStats()
+                RunInfoStatesEnum.GLOBAL_STATS -> showAverageSpeedKm()
+                RunInfoStatesEnum.AVERAGE_SPEED_KM -> showDurationKm()
+                RunInfoStatesEnum.DURATION_KM -> showDistanceSegment()
+                RunInfoStatesEnum.DISTANCE_SEGMENT -> showDurationSegment()
+                RunInfoStatesEnum.DURATION_SEGMENT -> showAverageSpeedSegment()
+                RunInfoStatesEnum.AVERAGE_SPEED_SEGMENT -> showPathDrawn()
+            }
+        }
+    }
+
+    /**
+     * Helper functions to defined the transitions made by clicking on left transition button.
+     */
+    private fun leftTransitionButton() {
+        this.changeLeftButton.setOnClickListener {
+            when (currentStateView) {
+                RunInfoStatesEnum.PATH_DRAWN -> showAverageSpeedSegment()
+                RunInfoStatesEnum.GLOBAL_STATS -> showPathDrawn()
+                RunInfoStatesEnum.AVERAGE_SPEED_KM -> showGlobalStats()
+                RunInfoStatesEnum.DURATION_KM -> showAverageSpeedKm()
+                RunInfoStatesEnum.DISTANCE_SEGMENT -> showDurationKm()
+                RunInfoStatesEnum.DURATION_SEGMENT -> showDistanceSegment()
+                RunInfoStatesEnum.AVERAGE_SPEED_SEGMENT -> showDurationSegment()
+            }
+        }
+    }
```

Comment:
It might be easier to understand and maintain by having an index that is incremented/ decremented that call the corresponding fragment. Here is one way to do it:

```kotlin
private enum class RunInfoStatesEnum (
     val show: () -> Unit
) {
     PATH_DRAWN({showPathDrawn()}),
     GLOBAL_STATS({showGlobalStats()}),
}
```

then in the function doing something like:

```kotlin
index = Math.floorMod(index + 1,  RunInfoStatesEnum.values().size)
RunInfoStatesEnum.values()[index].show()
```

and vice virsa for the other button